src/Controller/BackendCsvImportController.php

```diff
@@ -0,0 +1,223 @@
+ */
+
+namespace Contao\CoreBundle\Controller;
+
+use Contao\CoreBundle\Exception\RedirectResponseException;
+use Contao\CoreBundle\Framework\ContaoFrameworkInterface;
+use Contao\DataContainer;
+use Contao\FileUpload;
+use Contao\Message;
+use Doctrine\DBAL\Connection;
+use Contao\CoreBundle\Util\CsvImportUtil;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Symfony\Component\Security\Core\Authentication\Token\Storage\TokenStorageInterface;
+
+/**
+ * Handles the backend CSV import.
+ *
+ * @author Kamil Kuzminski <https://github.com/qzminski>
+ */
+class BackendCsvImportController
+{
+    /**
+     * @var Connection
+     */
+    private $connection;
+
+    /**
+     * @var ContaoFrameworkInterface
+     */
+    private $framework;
+
+    /**
+     * @var RequestStack
+     */
+    private $requestStack;
+
+    /**
+     * @var TokenStorageInterface
+     */
+    private $tokenStorage;
+
+    /**
+     * Constructor.
+     *
+     * @param Connection               $connection
+     * @param ContaoFrameworkInterface $framework
+     * @param RequestStack             $requestStack
+     * @param TokenStorageInterface    $tokenStorage
+     */
+    public function __construct(
+        Connection $connection,
+        ContaoFrameworkInterface $framework,
+        RequestStack $requestStack,
+        TokenStorageInterface $tokenStorage
+    ) {
+        $this->connection   = $connection;
+        $this->framework    = $framework;
+        $this->requestStack = $requestStack;
+        $this->tokenStorage = $tokenStorage;
+    }
+
+    /**
+     * Generate the import view for list wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws \Exception
+     * @throws RedirectResponseException
+     */
+    public function importListWizard(DataContainer $dc)
+    {
+        $uploader = $this->getDefaultUploader();
+
+        if ($uploader === null) {
+            return '';
+        }
+
+        $csvImport = new CsvImportUtil(
+            $this->connection,
+            new Message(),
+            $this->framework,
+            $this->requestStack->getCurrentRequest(),
+            $uploader
+        );
+
+        $csvImport->setCallback(function (array $data, array $row) {
+            return array_merge($data, $row);
+        });
+
+        if ($csvImport->isFormSubmitted()) {
+            $csvImport->run($dc->table, 'listitems', $dc->id);
+        }
+
+        return $csvImport->generate($GLOBALS['TL_LANG']['MSC']['lw_import'][0]);
+    }
+
+    /**
+     * Generate the import view for option wizard
+     *
+     * @param DataContainer $dc
+     *
+     * @return string
+     *
+     * @throws \Exception
+     * @throws RedirectResponseException
+     */
+    public function importOptionWizard(DataContainer $dc)
+    {
+        $uploader = $this->getDefaultUploader();
+
+        if ($uploader === null) {
+            return '';
+        }
+
+        $csvImport = new CsvImportUtil(
```

Comment:
I am not sure because then I would need to create the method that accepts at least 5 arguments (DataContainer, field name, data callback, submit label and allowed separators).